src/main/resources/common-services/CDAP/5.0.0/package/scripts/master.py

```diff
@@ -57,6 +60,62 @@ def start(self, env, upgrade_type=None):
+                                             if (output.startswith('[') and output.endswith(']')):
+                                                 if (output != '[]'):
+                                                     print(output)
+                                                     result = output[1:len(output)-1]
+                                                     # execute get /cdap/twill/<name>/instances/<last_output and save its output
+                                                     print(params.cdap_zookeeper_quorum)
+                                                     print(service)
+                                                     print(result)
+                                                     print('here executing the commands')
+                                                     print("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " get /twill/" + service + "/instances/" + result + " > /tmp/twill_" + service + "_instances_output")
+                                                     Execute("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " get /twill/" + service + "/instances/" + result + " > /tmp/twill_" + service + "_instances_output", user='zookeeper')
+                                                     status, output = commands.getstatusoutput("tail -n1 /tmp/twill_" + service + "_instances_output")
+                                                     if status == 0:
+                                                         print(output)
+                                                         data = json.loads(output)
+                                                         container_id = data['data']['containerId']
+                                                         container_values = container_id.split('_')
+                                                         applicationId = "application_" + container_values[2] + "_" + container_values[3] 
+                                                         kinit_cmd = params.kinit_cmd_master
+                                                         kill_application_cmd = format("{kinit_cmd} yarn application -kill " + applicationId)
+                                                         print(kill_application_cmd)
+                                                         Execute(kill_application_cmd, user='cdap')
+                                                         print("applicationId : " + applicationId)
+                                                     Execute("rm -rf /tmp/twill_" + service + "_instances_output", user='zookeeper') 
+                                         Execute("rm -rf /tmp/twill_" + service + "_instances", user='zookeeper') 
+                    Execute("rm -rf /tmp/twill_master_services", user='zookeeper')                  
+        Execute("rm -rf /tmp/election_master_services", user='zookeeper')             
```

Comment:
- Remove debug 'print' statements. 

- What is the criteria/reason for sleeping for '60' seconds ?  This will add delay when user stops/restarts CDAP.

- comands.getstatusoutput() is depreceated API. Use subprocess module

- I guess the assumption here is  that this script will be called with 'root' user privileges. Can this be avoided ?

Reply:
1 .remove the print statements, couple of them remain for printing the error message
used suprocess module for executing the command